eventuate-local-java-test-util/src/main/java/io/eventuate/local/testutil/CustomDBCreator.java

```diff
@@ -23,17 +28,34 @@ public CustomDBCreator(String dataFile, String dataSourceURL, String driverClass
     this.rootUserPassword = rootUserPassword;
   }
 
-  public void create() {
-    DataSource dataSource = DataSourceBuilder
+  public void create(Optional<SqlScriptEditor> editor) {
+    dataSource = DataSourceBuilder
             .create()
             .url(dataSourceURL)
             .driverClassName(driverClassName)
             .username(rootUserName)
             .password(rootUserPassword)
             .build();
 
-    Resource resource = new ClassPathResource(dataFile);
-    ResourceDatabasePopulator databasePopulator = new ResourceDatabasePopulator(resource);
-    databasePopulator.execute(dataSource);
+    jdbcTemplate = new JdbcTemplate(dataSource);
+
+    List<String> sqlList = loadSqlScriptAsListOfLines(dataFile);
+    if (editor.isPresent()) {
```

Comment:
isPresent() and get() are best avoided.

In this case use editor.ifPresent(e -> ... ) or editor.map(e -> e.edit(sqlList)).orElse(sqlList))